Tasks/CmdLine/cmdline.ts

```diff
@@ -10,9 +10,33 @@ async function run() {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
         // Get inputs.
-        let failOnStderr = tl.getBoolInput('failOnStderr', false);
-        let script: string = tl.getInput('script', false) || '';
-        let workingDirectory = tl.getPathInput('workingDirectory', /*required*/ true, /*check*/ true);
+        let input_failOnStderr = tl.getBoolInput('failOnStderr', false);
+        let input_workingDirectory = tl.getPathInput('workingDirectory', /*required*/ true, /*check*/ true);
+        let input_filePath: string;
+        let input_arguments: string;
+        let input_script: string;
+        let input_targetType: string = tl.getInput('targetType') || '';
+        if (input_targetType.toUpperCase() == 'FILEPATH') {
+            input_filePath = tl.getPathInput('filePath', /*required*/ true);
+            if (!tl.stats(input_filePath).isFile()) {
+                throw new Error(tl.loc('JS_InvalidFilePath', input_filePath));
+            }
+
+            input_arguments = tl.getInput('arguments') || '';
+        }
+        else {
+            input_script = tl.getInput('script', false) || '';
+        }
+
+        // Generate the script contents.
+        let contents: string;
+        if (input_targetType.toUpperCase() == 'FILEPATH') {
+            contents = `. '${input_filePath.replace("'", "''")}' ${input_arguments}`.trim();
```

Comment:
OK I played with this a little. I am planning to change it to only dot source if it ends with .sh (otherwise get permission issue if not +x). Dot sourcing a binary errors (attempts to interpret as shell script).